### ISSUE TITLE
::template(true) not working with connectedCallback

### ISSUE THREAD

**Issue by progelio** [x-tag/core-2#9]
xtag.create('x-test', class extends XTagElement {
            connectedCallback() { } //if I remove this, then it renders.
            name() { return "My Test Tag" }
            "::template(true)" () {
                return `<h2>I am ${this.name()}</h2>`
            }
        })

**Comment by ghost** [comment]
@csuwildcat I just plugged this into jasmine and it's working as described by @progelio

**Comment by progelio** [comment]
I don't know what jasmine is. I will find out. Try this:

<!DOCTYPE html>
<html>
<head>
    <meta charset="utf-8" />
    <title></title>
    <script src="x-tag-polyfilled.js"></script>
</head>
<body>
    <x-test></x-test>

    <script>
        xtag.create('x-test', class extends XTagElement {
            connectedCallback() { } //if I remove this, then it renders.
            name() { return "My Test Tag" }
            "::template(true)" () {
                return `<h2>I am ${this.name()}</h2>`
            }
        });
    </script>
</body>
</html>

**Comment by ghost** [comment]
If you have codepen I could provide a link to it with the test

**Comment by progelio** [comment]
Ok, just made a codepen account. Here it is:

https://codepen.io/progelio/pen/VxMgme

**Comment by csuwildcat** [comment]
The template extension is a class mixin that uses the connectedCallback, so
I'm guessing if you call super() in the custom element's connectedCallback
declaration, it will work as intended.

**Comment by ghost** [comment]
Got the error below.

SyntaxError: super() is only valid in derived class constructors.

**Comment by progelio** [comment]
this.render() works instead of super(). But, I think the presence of "connectedCallback" shouldn't interfere with "::template(true)"

**Comment by ghost** [comment]
![image](https://user-images.githubusercontent.com/11668353/39675546-4c688694-5122-11e8-9923-4ae4bf8a75bd.png)
here is a image tof he jasmine stack trace, would you like a pr with the Jasmine @csuwildcat ?

**Comment by ghost** [comment]
thanks @progelio.

**Comment by progelio** [comment]
Not a problem. This is actually a great project. Awesome work.

**Comment by ghost** [comment]
Jasmine Spec list image:  
![image](https://user-images.githubusercontent.com/11668353/39675605-18cfee66-5123-11e8-9afa-8c5cae929600.png)

**Comment by ghost** [comment]
@csuwildcat could `this.render()` be kickstarting the behavior by initiating the promise?

**Comment by ghost** [comment]
https://codepen.io/kipomaha/pen/XqeGNd

here is a link to a codepen I created with a codepen jasmine template
@progelio

**Comment by ghost** [comment]
for some reason its not passing the specs like in the jasmine image above

**Comment by ghost** [comment]
If you go to settings on the codepen you can see the dependencies by clicking the javascript tab when the settings screen  pops up

**Comment by ghost** [comment]
okay @progelio @csuwildcat 
I got the pen working here => https://codepen.io/kipomaha/pen/XqeGNd  [at the same pen url]

**Comment by ghost** [comment]
you can fork if you want their is a total of three errors

**Comment by csuwildcat** [comment]
I modded the original pen that had the error with the super call and it works: https://codepen.io/csuwldcat/pen/QrJrRg

**Comment by csuwildcat** [comment]
If you override a function, you must call super.FN_NAME(), so that the previous version of the function in the prototype chain is also called.

**Comment by ghost** [comment]
Nice did you want that to be put on the jasmine for the docs?

**Comment by ghost** [comment]
lol...I just looked in to super more I forgot you can use it like that duh...`super.funct()`....probably good that I don't use it much. Thanks for the information

**Comment by csuwildcat** [comment]
Yeah, we should probably document that functions that are extended require
this.

On a more general point: I want to get more of your contributions in, I
just been trying to find a way of taking some as smaller pulls that I can
digest easier. This is more about all the outstanding ones, not this exact
issue.

**Comment by ghost** [comment]
It's not a big deal, I stopped making PR's, because I figured you'd get to it when you got to it. 👍
